### PR TITLE
Null checks added

### PR DESCRIPTION
**Pull request by VishwaRajV**
Added Null checks in ef-coap.c to avoid segmentation faults

### CODE REVIEW

**Comment by HoratiuVultur** [comment]
Hi, can you say which was the command that generated the segmentation faults?

**Comment by VishwaRajV** [comment]
> Hi, can you say which was the command that generated the segmentation faults?

Hi,
 It was not a particular command. Segmentation fault occured while doing fuzz tests. So some random frame generated a SIGSEV error at the NULL checks that was added .

**Comment by HoratiuVultur** [comment]
Don't you have a log or something that shows all the generated commands at run-time? Because then you can extract the command that generated the random frame.
I am not saying it is not an issue but it would be nice to understand what is the command to see if other frames types are having the same issue.

**Comment by VishwaRajV** [comment]
Hi Horatiu ,

Sorry for the delay, took some time to analyze the actual problem that causes the Segmentation fault.

So , the random frame that got generated was :
1.  eth dmac 1 smac 2 ipv4 sip 1.0.0.2 dip 1.0.0.1 dip 1.0.0.1 udp sport 1000 dport 5683 coap ver 1 type 0 code 0.03 msgid                              0x5233 token 0123456789ABCD01 coap-opt num 11 val ascii c coap-opt num . val hex 8c

    The command is the coap-opt which calls the function option_fill_details() which tries to find the fields "num" and "val".
    In the above case , after the "num" field  "." is given which is invalid.
    So Segmentation fault occurs when it tries to read the value from the object fnum->val in : https://github.com/microchip-ung/easyframes/blob/0d1c8c257e29c8e5594082d6f38c9271c12bcfbf/src/ef-coap.c#L189

2. eth dmac 1 smac 2 ipv4 sip 1.0.0.2 dip 1.0.0.1 dip 1.0.0.1 udp sport 1000 dport 5683 coap ver 1 type 0 code 0.03 msgid                              0x5233 token 0123456789ABCD01 coap-opt num 11 val ascii c coap-opt num 0

   This is also similar to the first case , just that error occurs in : https://github.com/microchip-ung/easyframes/blob/0d1c8c257e29c8e5594082d6f38c9271c12bcfbf/src/ef-coap.c#L186
  This is because the frame is basically without "val" field , so when it tries to read fval->val Segmentation fault occurs.

**Comment by HoratiuVultur** [comment]
Thanks for the detail explanation and for finding the commands to reproduce the issue.
I have looked a little bit through the code and maybe easier would be to do like the following:

`diff --git a/src/ef-coap.c b/src/ef-coap.c`
`index a43d87d..50dd5f0 100644`
`--- a/src/ef-coap.c`
`+++ b/src/ef-coap.c`
`@@ -178,7 +178,7 @@ static int options_fill_defaults(struct frame *f, int stack_idx) {`
`     field_t *fnum = find_field(hdr, "num");`
`     field_t *fval = find_field(hdr, "val");`
`-    if (!fnum) { `
`+    if (!fnum->val || !fval->val) { `
`         return 0;`
`     }`

**Comment by VishwaRajV** [comment]
Thank you  for the suggestion, 
One concern is that the check is added inside !fnum check,
```
if(!fnum){
   if (!fnum->val || !fval->val) { the problem is that , the check is useful only when fnum is not NULL
     return 0; 
   }
}
```

But the segmentation occurs when fnum->val or fval->val is NULL, I have also further tested it out 
So It could be changed to the following:

```
if(!fnum){
     return 0; 
   }
if(!fnum->val ||!fval->val){
    return 0;
}
```

We could have two separate checks.

**Comment by HoratiuVultur** [comment]
I didn't mean to put the check inside 
````
if (!fnum) {
    return 0;
}
````
but to replace that code with
````
if (!fnum->val || !fval->val) {
    return 0;
}
````
Because `fnum` can't be NULL(if I read correctly the code) so there is no point to have that check anymore.

**Comment by VishwaRajV** [comment]
Yes okay Horatiu, Understood now. 
I have updated the branch now with the said suggestion.
The PR is also automatically updated.
Thank you

**Comment by HoratiuVultur** [comment]
Please squash those 2 commits into one single commit and add a single comment. Then I think it is ready to be merged.

**Comment by VishwaRajV** [comment]
Yes updated into one commit. 
Thank you

File: src/ef-coap.c [hunk 1]
```diff
@@ -178,11 +178,11 @@ static int options_fill_defaults(struct frame *f, int stack_idx) {
     field_t *fnum = find_field(hdr, "num");
     field_t *fval = find_field(hdr, "val");
 
-    if (!fnum) {
+    if (!fnum->val || !fval->val) {
         return 0;
     }
 
-    if (fval){
+    if (fval) {
```
Comment (HoratiuVultur) [inline src/ef-coap.c]: This change is just a style change so it is not needed in this commit.

**Review by HoratiuVultur — CHANGES_REQUESTED** [review]

File: src/ef-coap.c (hunk 1 above)
Comment (VishwaRajV) [reply src/ef-coap.c]: Yes okay , Updated now .
Thank you

**Comment by HoratiuVultur** [comment]
This looks good to me. Thanks for doing all this!

**Pull request merged**